openshift-knative-operator/pkg/monitoring/monitoring.go

```diff
@@ -0,0 +1,115 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	mf "github.com/manifestival/manifestival"
+	"github.com/openshift-knative/serverless-operator/openshift-knative-operator/pkg/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/operator/pkg/apis/operator/v1alpha1"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	EnableMonitoringEnvVar       = "ENABLE_SERVING_MONITORING_BY_DEFAULT"
+	EnableMonitoringLabel        = "openshift.io/cluster-monitoring"
+	ObservabilityCMName          = "observability"
+	ObservabilityBackendKey      = "metrics.backend-destination"
+	OpenshiftMonitoringNamespace = "openshift-monitoring"
+)
+
+const (
+	prometheusRoleName        = "knative-serving-prometheus-k8s"
+	prometheusClusterRoleName = "rbac-proxy-metrics-prom"
+)
+
+func SetupServingMonitoring(ctx context.Context, api kubernetes.Interface, ks *v1alpha1.KnativeServing) error {
```

Comment:
It'd be great to name this `ReconcileServingMonitoring` to signify the fact that this runs per reconcile loop. Likewise I'd rename `setMonitoringLabelToNamespace` to `reconcileMonitoringLabelOnNamespace` to signify the same thing.